src/components/views/settings/KeyBackupPanel.js

```diff
@@ -227,11 +227,17 @@ export default class KeyBackupPanel extends React.PureComponent {
                         "Backup has a signature from <verify>unknown</verify> device with ID %(deviceId)s.",
                         { deviceId: sig.deviceId }, { verify },
                     );
-                } else if (sig.device.getFingerprint() === MatrixClientPeg.get().getDeviceEd25519Key()) {
+                } else if (sig.device.getFingerprint() === MatrixClientPeg.get().getDeviceEd25519Key() && sig.valid) {
```

Comment:
I'd suggest moving the `sig.valid` portion of each of these to the beginning, so that it matches the later branches below it. Also, extracting `sig.device.getFingerprint() === MatrixClientPeg.get().getDeviceEd25519Key()` to `const fromThisDevice = ...` above the long `if` list would help readability.